frontendservice/src/views/SignUp.vue

```diff
@@ -0,0 +1,172 @@
+    </div>
+    <p>{{passwordAlert}}</p>
+    <a class="button is-primary" @click="createUser">Sign Up</a>
+    <p>{{firebaseAlert}}</p>
+    <div>
+      If you have already registered, go to <router-link to="/login">login</router-link>.
+    </div>
+    <div>
+      Or go back to <router-link to="/about">about</router-link>.
+    </div>
+  </div>
+</template>
+
+<script>
+import firebase from "firebase";
+export default {
+  name: "signUp",
+  data() {
+    return {
+      email: "",
+      password: "",
+      name: "",
+      isHidden: true,
+      nameAlert: "",
+      emailAlert: "",
+      passwordAlert: "",
+      firebaseAlert: ""
+    };
+  },
+  methods: {
+    toggle() {
+      var passwordRef = this.$refs.password;
+      if (passwordRef.type === "password") {
+        passwordRef.type = "text";
+      } else {
+        passwordRef.type = "password";
+      }
+      this.isHidden = !this.isHidden;
+    },
+    createUser: function() {
+      this.nameAlert = "";
+      this.emailAlert = "";
+      this.passwordAlert = "";
+      this.firebaseAlert = "";
+
+      //to check if the input email follows correct format like *@*.*
+      //eg - virtualqueues1@gmail.com is valid
+      var emailRegex = new RegExp(/\w+@\w+.\w+/g);
+      //to check if the input password contains any digit
+      //eg - xyz123 is valid
+      var passwordRegex1 = new RegExp(/[0-9]/g);
+      //to check if the input password contains any special character (!, @, #, %, &, _ or -)
+      //eg - abcde! is valid
+      var passwordRegex2 = new RegExp(/(!|@|#|%|&|_|-)/g);
+      var emailResult = this.email.match(emailRegex);
+      var passwordResult1 = this.password.match(passwordRegex1);
+      var passwordResult2 = this.password.match(passwordRegex2);
+
+      if (this.name.length <= 3 && this.name.length >= 0) {
+        this.nameAlert = "Name must be of atleast 3 characters.";
+      } else if (this.name.length > 30) {
+        this.nameAlert = "Name can have atmost 30 characters only.";
+      }
+      if (this.email.length == 0 || !emailResult) {
+        this.emailAlert = "Invalid email address entered.";
+      }
+      if (this.password.length >= 0 && this.password.length < 6) {
+        this.passwordAlert = "Password length must be greater than 5 characters.";
+      } else if (!passwordResult1 && !passwordResult2) {
+        this.passwordAlert = "Password must contain a digit or special character (!, @, #, %, &, _, -).";
+      }
+      if (this.nameAlert == "" && this.emailAlert == "" && this.passwordAlert == "") {
+        firebase
+          .auth()
+          .createUserWithEmailAndPassword(this.email, this.password)
+          .then(
+            user => {
+              firebase
+                .auth()
+                .currentUser.updateProfile({
+                  displayName: this.name
+                })
+                .then(function() {
+                })
```

Comment:
Curious, why is this empty? Are we planning to add something here in a follow-up PR? If not, can this be removed?

Reply:
Removed.